solr/core/src/java/org/apache/solr/search/facet/FacetRequest.java

```diff
@@ -287,24 +288,23 @@ public static void createGraphField(FacetRequest.Domain domain, Map<String,Objec
        * Creates a Query that can be used to recompute the new "base" for this domain, relative to the
        * current base of the FacetContext.
        */
-      public Query createDomainQuery(FacetContext fcontext) throws IOException {
-        final SolrConstantScoreQuery fromQuery = new SolrConstantScoreQuery(fcontext.base.getTopFilter());
+      public Query createDomainQuery(FacetContext fcontext) {
+        final ConstantScoreQuery fromQuery = new ConstantScoreQuery(fcontext.base.makeQuery());
```

Comment:
Do we need to wrap this in a constant score query?

Reply:
Refactored it so `fromQuery` is not wrapped in a constant score query given we are using makeQuery which is a constant scoring query